server/routes/me.js

```diff
@@ -1010,6 +1010,12 @@ router.get('/courses', async (req, res) => {
     }
   }
 
+  if (courses) {
+    for (let i = 0; i < courses.length; i++) {
+      courses[i]['getCert'] = courses[i].Subjects && courses[i].Subjects.length > 0;
```

Comment:
cái đoạn này nó có các môn nhưng chưa có điểm thì nó vẫn có get cert à ?

Reply:
> cái đoạn này nó có các môn nhưng chưa có điểm thì nó vẫn có get cert à ?

đoạn check điểm đã có ở route post /certificate rồi ạ, cần phải validate ngay taị front end nữa ạ ?

Reply:
vẫn phải check vì chưa học gì nó đã hiện ra nút đấy rồi nghe không hợp lý lắm

Reply:
Vậy có thật sự cần không ạ. Vì hiện tại các course đang chỉ lưu subject dạng id như này
```
Subjects: ["bbaa9a96-49b6-4da6-b2ca-ba9b9ef4a0e1", "d261df6d-ae8b-4c40-bb04-ad7e40840f1c"]
```
Giờ muốn check điểm của một course là sẽ cần query vào 2 nơi score và subject để tìm ra điểm. Một course thì đã vậy bây giờ nếu `n` course liên tục vậy thì thực hiện query lồng query và lồng vào for nữa em chỉ muốn hỏi xem có cần thiết thật sự không. Nếu cần thiết thì em sẽ code còn không em nghĩ nên chuyển nút getCert vào trong course detail lúc này mình chỉ cần check điểm của 1 thằng course sẽ giống hệt route phần get course detail bây giờ và thêm phần check điểm nữa là được ạ.

Reply:
![image](https://user-images.githubusercontent.com/49506301/76382983-fed27c80-638c-11ea-8fb9-bd2513d0776c.png)
đây là list course ạ

Reply:
```
const resSubjects = await network.query(networkObj, 'GetSubjectsOfCourse', courseId);
  const resScores = await network.query(networkObj, 'GetScoresOfStudent', [username, courseId]);
  if (!resSubjects.success || !resScores.success) {
    return res.status(500).json({
      success: false,
      msg: 'query chaincode error!'
    });
  }
  let listSubjects = JSON.parse(resSubjects.msg) ? JSON.parse(resSubjects.msg) : [];
  let listScores = JSON.parse(resScores.msg) ? JSON.parse(resScores.msg) : [];

  for (let i = 0; i < listScores.length; i++) {
    for (let j = 0; j < listSubjects.length; j++) {
      if (listScores[i].SubjectID === listSubjects[j].SubjectID) {
        let subject = listSubjects[j];
        subject['score'] = listScores[i].ScoreValue;
        listSubjects[j] = subject;
      }
    }
  }
```

đây là code để get score của các subject trong 1 course

Reply:
vậy  n course t sẽ phải lồng for chạy n lần cái này ạ

Reply:
ok cái này tạm pending, đợi sửa chaincode có thể query dễ dàng bằng ID thì sẽ tiến hành refactor